ecommerce/extensions/api/v2/views/enterprise.py

```
@@ -84,6 +91,29 @@ def get(self, request):
         )
         return Response(data=enterprise_catalogs)
 
+    def retrieve(self, request, **kwargs):
+        endpoint_request_url = urlparse(request.build_absolute_uri())._replace(query=None).geturl()
+        try:
+            catalog = get_enterprise_catalog(
+                site=request.site,
+                enterprise_catalog=kwargs.get('enterprise_catalog_uuid'),
+                limit=request.GET.get('limit', DEFAULT_CATALOG_PAGE_SIZE),
+                page=request.GET.get('page', '1'),
+                endpoint_request_url=endpoint_request_url
+            )
+        except (ConnectionError, SlumberHttpBaseException, Timeout) as exc:
+            logger.exception(
+                'Unable to retrieve catalog for enterprise customer! customer: %s, Exception: %s',
+                kwargs.get('enterprise_catalog_uuid'),
+                exc
+            )
+            return Response(
+                {'error': 'Unable to retrieve enterprise catalog'},
```

i think we can surface the exception message here as well